### YamlMime:PythonClass
uid: azure.cognitiveservices.personalizer.models.personalizer_error.PersonalizerError
name: PersonalizerError
fullName: azure.cognitiveservices.personalizer.models.personalizer_error.PersonalizerError
module: azure.cognitiveservices.personalizer.models.personalizer_error
inheritances:
- msrest.serialization.Model
summary: 'The error object.


  All required parameters must be populated in order to send to Azure.'
constructor:
  syntax: PersonalizerError(**kwargs)
  parameters:
  - name: code
    description: 'Required. High level error code. Possible values include:

      ''BadRequest'', ''ResourceNotFound'', ''InternalServerError'''
    types:
    - '[str](https://docs.python.org/3.6/library/stdtypes.html#str) or <xref:azure.cognitiveservices.personalizer.models.ErrorCode>'
  - name: message
    description: 'Required. A message explaining the error reported by the

      service.'
    types:
    - '[str](https://docs.python.org/3.6/library/stdtypes.html#str)'
  - name: target
    description: Error source element.
    types:
    - '[str](https://docs.python.org/3.6/library/stdtypes.html#str)'
  - name: details
    description: 'An array of details about specific errors that led to this

      reported error.'
    types:
    - '[list](https://docs.python.org/3.6/library/stdtypes.html#list)[<xref:azure.cognitiveservices.personalizer.models.PersonalizerError>]'
  - name: inner_error
    description: Finer error details.
    types:
    - <xref:azure.cognitiveservices.personalizer.models.InternalError>
